Add support for VisionOS (WIP) (#8027)
libsdl-org/SDL: CMakeLists.txt

# See docs/release_checklist.md
project(SDL3 LANGUAGES C CXX VERSION "3.0.0")

if (CMAKE_SYSTEM_NAME MATCHES "visionOS")
  set(VISIONOS ON)
else()
  set(VISIONOS OFF)
endif()

> 

Reviewer: Please move this to [here](https://github.com/libsdl-org/SDL/blob/8adab0b88ab1c0aa69242fffc48afdba085b1dc4/cmake/sdlplatform.cmake#L8-L17).

#  so we'll just use libusb when it's available. libusb does not support iOS,
#  so we default to yes on iOS.
#  TODO: Windows can support libusb, the hid.c file just depends on Unix APIs

> if((WINDOWS AND NOT WINDOWS_STORE) OR IOS OR TVOS OR ANDROID OR VISIONOS)

Reviewer 2: For consistency, can you move this after TVOS?
```
if((WINDOWS AND NOT WINDOWS_STORE) OR IOS OR TVOS OR VISIONOS OR ANDROID)
```

set_option(SDL_DUMMYAUDIO          "Support the dummy audio driver" ON)
set_option(SDL_DUMMYVIDEO          "Use dummy video driver" ON)
dep_option(SDL_IBUS                "Enable IBus support" ON ${UNIX_SYS} OFF)

> set_option(SDL_OPENGL              "Include OpenGL support" ON "VISIONOS" OFF)

Reviewer 2: I think `"VISIONOS"` is evaluating to true, since it's a string. This is what's causing the build errors - OpenGL is disabled for all platforms. Maybe try just `VISIONOS`?

Reviewer: Assuming enable opengl by default for all non-visionos, and force disable for vision (with no way to re-enable it)

```suggestion
dep_option(SDL_OPENGL              "Include OpenGL support" ON "NOT VISIONOS" OFF)
```

> set_option(SDL_OPENGLES            "Include OpenGL ES support" ON "VISIONOS" OFF) 

Reviewer: Likewise
```suggestion
dep_option(SDL_OPENGLES            "Include OpenGL ES support" ON "NOT VISIONOS" OFF) 
```

set_option(SDL_OFFSCREEN           "Use offscreen video driver" ON)
option_string(SDL_BACKGROUNDING_SIGNAL "number to use for magic backgrounding signal or 'OFF'" OFF)
option_string(SDL_FOREGROUNDING_SIGNAL "number to use for magic foregrounding signal or 'OFF'" OFF)

> set_option(SDL_HIDAPI              "Enable the HIDAPI subsystem" ON "VISIONOS" OFF)

Reviewer: ```suggestion
dep_option(SDL_HIDAPI              "Enable the HIDAPI subsystem" ON "NOT VISIONOS" OFF)
```

Reviewer: This one's left.

          if(PC_ICONV_FOUND)
            sdl_link_dependency(iconv LIBS Iconv::Iconv CMAKE_MODULE Iconv PKG_CONFIG_SPECS iconv)
          else()
            if (VISIONOS)

>               sdl_link_dependency(iconv LIBS iconv CMAKE_MODULE Iconv PKG_CONFIG_LIBS iconv)

Reviewer: Doesn't cmake's FindIconv.cmake work for visions?

Author: It didn't appear to. I reported it CMake (https://gitlab.kitware.com/cmake/cmake/-/issues/25161) and will revert this change.
